source/dub/dependency.d

```diff
@@ -678,6 +719,34 @@ unittest {
 	assert(Dependency("^1.2").versionSpec == "~>1.2"); // equivalent; prefer ~>
 }
 
+/**
+	Represents an SCM repository.
+*/
+struct Repository
+{
+	private string m_remote;
+
+	/**
+		Returns:
+			Repository URL or path.
+	*/
+	@property string remote() @nogc nothrow pure @safe
+	in { assert(m_remote !is null); }
+	body
+	{
+		return m_remote;
+	}
+
+	/**
+		Returns:
+			Whether the repository was initialized with an URL or path.
+	*/
+	@property bool empty() const @nogc nothrow pure @safe
+	{
+		return m_remote is null;
```

Comment:
`return !m_remote.length;` or `return m_remote.empty;` might be better because empty will currently be false if Repository was initialized with `""`

It will slightly change behavior as currently an `""` repository will actually serialize and deserialize properly as empty string instead of not being there however I think such a case should rather be handled as edge case in the parser and not be allowed in the first place.

Reply:
Added `enforce`.